version: "3.7"
x-logging: &logging
  logging:
    driver: loki
    options:
      loki-url: "http://localhost:3100/loki/api/v1/push"
services:
  web:
    image: ${CI_REGISTRY_IMAGE:-i69}/${IMAGE:-web}:${SANITIZED_BRANCH:-latest}
    build:
      context: .
      dockerfile: ./docker/django/Dockerfile.dev
      args:
        UID: ${UID:-10600}
        GID: ${GID:-10600}
        UNAME: ${UNAME:-i69}
    container_name: web
    ports:
      - "9082:80"
    hostname: web
    volumes:
      - media_volume:/home/i69/media
      - ./:/home/i69
    env_file:
      - ${ENV_FILE}
    depends_on:
      - postgres
      - redis
    networks:
      - i69_net

  postgres:
    image: postgres:13.6-alpine3.15
    container_name: postgres
    hostname: postgres
    restart: on-failure
    ports:
      - "5432:5432"
    volumes:
      - postgres_data_recover:/var/lib/postgresql/data
    env_file:
      - ${ENV_FILE}
    networks:
      - i69_net

  redis:
    image: ${CI_REGISTRY_IMAGE:-i69}/${IMAGE:-redis}:${SANITIZED_BRANCH:-latest}
    build:
      context: .
      dockerfile: ./docker/redis/Dockerfile
    container_name: redis
    hostname: redis
    volumes:
      - redis_data:/data
    networks:
      - i69_net

networks:
  i69_net:

volumes:
  postgres_data_recover:
    external: true
    name: postgres_postgres_data_recover
  redis_data:
  media_volume:
    external: true
    name: i69app_storage
